pytest/tests/test_sudo.py

```diff
@@ -0,0 +1,634 @@
+        5. Command is successful
+        6. Command is successful
+    :customerscenario: False
+    """
+    provider.user("USER-1").add()
+    provider.sudorule("lowercase").add(user="user-1", host="ALL", command="/bin/less")
+    provider.sudorule("uppsercase").add(user="USER-1", host="ALL", command="/bin/more")
+    client.fs.write("/root/test", "test")
+
+    client.sssd.common.sudo()
+    client.sssd.domain["case_sensitive"] = "false"
+    client.sssd.start()
+
+    assert client.auth.sudo.list(
+        "user-1", "Secret123", expected=["(root) /bin/less", "(root) /bin/more"]
+    ), "Sudo list failed!"
+    assert client.auth.sudo.run("user-1", "Secret123", command="/bin/less /root/test"), "Sudo command failed!"
+    assert client.auth.sudo.run("user-1", "Secret123", command="/bin/more /root/test"), "Sudo command failed!"
+
+    assert client.auth.sudo.list(
+        "USER-1", "Secret123", expected=["(root) /bin/less", "(root) /bin/more"]
+    ), "Sudo list failed!"
+    assert client.auth.sudo.run("USER-1", "Secret123", command="/bin/less /root/test"), "Sudo command failed!"
+    assert client.auth.sudo.run("USER-1", "Secret123", command="/bin/more /root/test"), "Sudo command failed!"
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopology.BareAD)
+@pytest.mark.topology(KnownTopology.BareIPA)
+@pytest.mark.topology(KnownTopology.BareLDAP)
+@pytest.mark.parametrize("sssd_service_user", ("root", "sssd"))
+@pytest.mark.require(
+    lambda client, sssd_service_user: ((sssd_service_user == "root") or client.features["non-privileged"]),
+    "SSSD was built without support for running under non-root",
+)
+def test_sudo__rules_refresh(client: Client, provider: GenericProvider, sssd_service_user: str):
+    """
+    :title: Sudo rules refresh works
+    :setup:
+        1. Create user "user-1"
+        2. Create sudorule to allow "user-1" run "/bin/ls on all hosts
+        3. Enable SSSD sudo responder
+        4. Set "entry_cache_sudo_timeout" to "2"
+        5. Start SSSD
+    :steps:
+        1. List sudo rules for "user-1"
+        2. Modify the rule to allow only "/bin/less" command
+        3. Wait until the cached rule is expired (3 seconds)
+        4. List sudo rules for "user-1"
+    :expectedresults:
+        1. User is able to run only /bin/ls
+        2. Rule was modified
+        3. Time passed
+        4. User is able to run only /bin/less
+    :customerscenario: False
+    """
+    u = provider.user("user-1").add()
+    r = provider.sudorule("test").add(user=u, host="ALL", command="/bin/ls")
+
+    client.sssd.common.sudo()
+    client.sssd.domain["entry_cache_sudo_timeout"] = "2"
+    client.sssd.start(service_user=sssd_service_user)
+
+    assert client.auth.sudo.list("user-1", "Secret123", expected=["(root) /bin/ls"]), "Sudo list failed!"
+    r.modify(command="/bin/less")
+    time.sleep(3)
+    assert client.auth.sudo.list("user-1", "Secret123", expected=["(root) /bin/less"]), "Sudo command failed!"
```

Comment:
Along with listing of correct sudo rules, try to test whether the user is actually able to run the allowed sudo command.